frontend/middleware_validatesubscription.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	"github.com/Azure/ARO-HCP/internal/api/subscription"
+)
+
+type SubscriptionStateMuxValidator struct {
+	cache *Cache
+}
```

Comment:
Suggestion...

If we added the cache to the `http.Server`'s base context in `NewFrontend` under a new `ContextKeyCache` constant, then this middleware wouldn't need its own struct (which is cleaner, in my opinion).  `MiddlewareValidateSubscriptionState` could just fetch the cache from the given request's context.

Reply:
That's a fair point, I'm not sure we would want that either.  Ok, so good as is.